src/main/resources/db/migration/V1__create_initial_schema.sql

```
@@ -42,3 +42,30 @@ CREATE TABLE conversation_read_state (
+    CONSTRAINT chk_connection_resolution CHECK (
+            (status = 'PENDING' AND resolved_at IS NULL)
+                OR
+            (status IN ('ACCEPTED', 'REJECTED') AND resolved_at IS NOT NULL)
+            ));
+
+CREATE UNIQUE INDEX uk_connection_users
+    ON connection (
+                   LEAST(request_sender_id, request_receiver_id),
+                   GREATEST(request_sender_id, request_receiver_id)
+        );
+CREATE INDEX idx_connection_incoming_pending ON connection(request_receiver_id) WHERE status = 'PENDING';
```

SQLFluff (4.2.2)

[error] CREATE INDEX should use CONCURRENTLY to avoid locking the table during the build.

(PG01)